bundle/apps/validate.go

```diff
@@ -0,0 +1,51 @@
+package apps
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+)
+
+type validate struct{}
+
+func (v *validate) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	var diags diag.Diagnostics
+	possibleConfigFiles := []string{"app.yml", "app.yaml"}
+	usedSourceCodePaths := make(map[string]string)
+
+	for key, app := range b.Config.Resources.Apps {
+		if _, ok := usedSourceCodePaths[app.SourceCodePath]; ok {
+			diags = append(diags, diag.Diagnostic{
+				Severity:  diag.Error,
+				Summary:   "Duplicate app source code path",
+				Detail:    fmt.Sprintf("app resource '%s' has the same source code path as app resource '%s'", key, usedSourceCodePaths[app.SourceCodePath]),
+				Locations: b.Config.GetLocations(fmt.Sprintf("resources.apps.%s.source_code_path", key)),
+			})
+		}
+		usedSourceCodePaths[app.SourceCodePath] = key
+
+		for _, configFile := range possibleConfigFiles {
+			cf := path.Join(app.SourceCodePath, configFile)
```

Comment:
Is this guaranteed to be relative?

Reply:
Yes, the app.yml has to be at the root of the source code path

Reply:
The call in the upload mutator suggests otherwise:

```go
appPath := strings.TrimPrefix(app.SourceCodePath, b.Config.Workspace.FilePath)
```

Reply:
Ah, you're right, thanks for catching this. Fixed and updated the test to catch this as well